gp2gp-translator/src/test/java/uk/nhs/adaptors/pss/translator/util/BloodPressureValidatorUtilTest.java

```diff
@@ -0,0 +1,132 @@
+    }
+
+    @Test
+    public void invalidSystolicBloodPressureCodeReturnsFalse() {
+        assertThat(BloodPressureValidatorUtil.isSystolicBloodPressure(INVALID_CODE)).isFalse();
+    }
+
+    @ParameterizedTest(name = "validDiastolicBloodPressure")
+    @MethodSource("validDiastolicBloodPressures")
+    public void validDiastolicBloodPressureCodeReturnsTrue(String diastolicCode) {
+        assertThat(BloodPressureValidatorUtil.isDiastolicBloodPressure(diastolicCode)).isTrue();
+    }
+
+    @Test
+    public void invalidDiastolicBloodPressureCodeReturnsFalse() {
+        assertThat(BloodPressureValidatorUtil.isDiastolicBloodPressure(INVALID_CODE)).isFalse();
+    }
+
+    private static Stream<Arguments> validBloodPressureTriples() {
+        return Stream.of(
+                generateBloodPressureTriples(HEADER_1, SYSTOLIC_1, DIASTOLIC_1),
+                generateBloodPressureTriples(HEADER_1, SYSTOLIC_2, DIASTOLIC_2),
+                generateBloodPressureTriples(HEADER_2, SYSTOLIC_1, DIASTOLIC_1),
+                generateBloodPressureTriples(HEADER_2, SYSTOLIC_2, DIASTOLIC_2),
+                generateBloodPressureTriples(HEADER_3, SYSTOLIC_1, DIASTOLIC_1),
+                generateBloodPressureTriples(HEADER_3, SYSTOLIC_2, DIASTOLIC_2),
+                generateBloodPressureTriples(HEADER_4, SYSTOLIC_3, DIASTOLIC_3),
+                generateBloodPressureTriples(HEADER_5, SYSTOLIC_4, DIASTOLIC_4),
+                generateBloodPressureTriples(HEADER_6, SYSTOLIC_5, DIASTOLIC_5))
+            .flatMap(Collection::stream)
+            .collect(Collectors.toList()).stream();
+    }
+
+    private static Stream<Arguments> validSystolicBloodPressures() {
+        return Stream.of(
+            Arguments.of("72313002"),
+            Arguments.of("120159016"),
+            Arguments.of("271649006"),
+            Arguments.of("406507015"),
+            Arguments.of("400974009"),
+            Arguments.of("1780182014"),
+            Arguments.of("407554009"),
+            Arguments.of("2159155011"),
+            Arguments.of("407556006"),
+            Arguments.of("2159157015"));
+    }
+
+    private static Stream<Arguments> validDiastolicBloodPressures() {
+        return Stream.of(
+            Arguments.of("1091811000000102"),
+            Arguments.of("2734671000000117"),
+            Arguments.of("271650006"),
+            Arguments.of("406508013"),
+            Arguments.of("400975005"),
+            Arguments.of("1780183016"),
+            Arguments.of("407555005"),
+            Arguments.of("2159156012"),
+            Arguments.of("407557002"),
+            Arguments.of("2159158013"));
+    }
+
+    private static ArrayList<Arguments> generateBloodPressureTriples(List<String> headers, List<String> systolics,
+        List<String> diastolics) {
+        var testCases = new ArrayList<Arguments>();
+        for (String header
+            : headers) {
```

Comment:
Maybe all of those fors could be in one line, like `for (String header : headers)`